helix-term/src/ui/editor.rs

```diff
@@ -509,11 +511,25 @@ impl EditorView {
             {
                 // ensure col is on screen
                 if let Some(highlight) = theme.find_scope_index_exact("ui.cursor.match") {
-                    return vec![(highlight, pos..pos + 1)];
+                    elements.push((highlight, pos..pos + 1));
+                }
+            }
+
+            // Highlight surrounding brace
+            if let Some(highlight) = theme.find_scope_index_exact("ui.cursor.surround") {
```

Comment:
I put this check first because I assume it would be faster than `find_matching_bracket_fuzzy`.